pyqtgraph/graphicsItems/PlotItem/PlotItem.py

```diff
@@ -1171,6 +1173,64 @@ def showAxis(self, axis, show=True):
     def hideAxis(self, axis):
         """Hide one of the PlotItem's axes. ('left', 'bottom', 'right', or 'top')"""
         self.showAxis(axis, False)
+        
+    def showAxes(self, selection, showValues=True, size=None):
+        """ 
+        Convenience method for quickly configuring axis settings.
+        
+        Parameters
+        ----------
+        selection: boolean or tuple of booleans (left, top, right, bottom)
+            Determines which AxisItems will be displayed.
+            A single boolean value will set all axes, 
+            so that ``showAxes(True)`` configures the axes to draw a frame.
+        showValues: optional, boolean or tuple of booleans (left, top, right, bottom)
+            Determines if values will be displayed for the ticks of each axis.
+            True value shows values for left and bottom axis (default).
+            False shows no values.
+            None leaves settings unchanged.
+            If not specified, left and bottom axes will be drawn with values.
+        size: optional, float or tuple of floats (width, height)
+            Reserves as fixed amount of space (width for vertical axis, height for horizontal axis)
+            for each axis where tick values are enabled. If only a single float value is given, it
+            will be applied for both width and height.
+        """
+        if selection is True: # shortcut: enable all axes, creating a frame
+            selection = (True, True, True, True)
+        elif selection is False: # shortcut: disable all axes
+            selection = (False, False, False, False)
+        if showValues is True: # shortcut: defaults arrangement with labels at left and bottom
+            showValues = (True, False, False, True)
+        elif showValues is False: # shortcut: disable all labels
+            showValues = (False, False, False, False)
+        elif showValues is None: # leave labelling untouched
+            showValues = (None, None, None, None)
+        if size is None:
+            size = (None, None)
+        elif not isinstance(size, collections.abc.Sized):
+            size = (size, size)
+        for idx, key in enumerate(('left','top','right','bottom')):
```

Comment:
this loop bugs me a bit, because there is a fair amount of almost-duplicated code; ...earlier when I commented on this, I missed the case where you could have `None`, I think I re-wrote this to handle that use-case accordingly.

```python
axes = ('left', 'top', 'right', 'bottom')
for sub_selection, showValue, axis in zip(selection, showValues, axes):
    if sub_selection is True:
        self.showAxis(axis)
    elif sub_selection is False:
        self.hideAxis(axis)
    else:
        pass
    if showValue is None:
        continue
    ax = self.getAxis(axis)
    if showValue:
        ax.setStyle(showValues=showValue)
        if axis in ('left', 'right') and size[0] is not None:
            ax.setWidth(size[0] if showValue else None)
        elif axis in ('top', 'bottom') and size[1] is not None:
            ax.setHeight(size[1] if showValue else None)
```

Reply:
Here's what I get if I try to sort out the duplicate checks, but try to keep the bypass for None values explicit. I also realized that setting size to None should revert back to auto-size, because otherwise there is no way to do that. 

``` python
all_axes = ('left','top','right','bottom')
        for show_axis, show_value, axis_key in zip(selection, showValues, all_axes):
            if show_axis is None:
                pass # leave axis display as it is.
            else:
                if show_axis: self.showAxis(axis_key)
                else        : self.hideAxis(axis_key)
                
            if show_value is None:
                pass # leave value display as it is.
            else:
                ax = self.getAxis(axis_key)
                ax.setStyle(showValues=show_value)
                if size is not False: # size adjustment is requested
                    if axis_key in ('left','right'):
                        if show_value: ax.setWidth(size[0])
                        else         : ax.setWidth( None )
                    elif axis_key in ('top', 'bottom'):
                        if show_value: ax.setHeight(size[1])
                        else         : ax.setHeight( None )
```